src/path.rs

```diff
@@ -298,6 +298,14 @@ fn path_get_path_core<S: AsRef<wstr>>(cmd: &wstr, pathsv: &[S]) -> GetPathResult
         return GetPathResult::new(test_path(cmd).err(), cmd.to_owned());
     }
 
+    if let Err(err) = test_path(cmd) {
```

Comment:
it doesn't make sense to call `test_path` here because the argument is not a path.
We only do this to to detect some cases where command name is too long regardless of the path, so `libc::access` should be enough.
But I'm not sure if it's even worth to have this extra code path.

Is prepending the first $PATH element really confusing in practice?

 >  it could be that the file exists at one of the proposed paths
 >  but simply can't be executed/accessed because the given path is
 >  too long.

I'd be surprised if that was true  - that would imply that it's
possible to actually create a file that is impossible to access/delete
with a naively implemented "rm"